generic/tests/pktgen_perf.py

```diff
@@ -0,0 +1,186 @@
+    record_list = params.get("record_list")
+    error_context.context("Init the VM, and try to login", logging.info)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=login_timeout)
+    session_serial = vm.wait_for_serial_login(timeout=timeout)
+
+    # print numa information on host and pinning vhost and vcpus to cpus
+    process.system_output("numactl --hardware")
+    process.system_output("numactl --show")
+    _pin_vm_threads(params.get("numa_node"))
+
+    # get parameter from dictionary
+    pktgen_stress_timeout = float(params.get("pktgen_test_timeout", 60))
+    shell_port = params.get("shell_port", "22")
+    username = params.get("username", "root")
+    password = params.get("password", "123456")
+    category = params.get("category")
+    pkt_size = params.get("pkt_size")
+    kvm_ver_chk_cmd = params.get("kvm_ver_chk_cmd")
+    guest_ver_cmd = params["guest_ver_cmd"]
+
+    # get qemu, guest kernel and kvm version info and write them into result
+    result_path = utils_misc.get_path(test.resultsdir, "pktgen_perf.RHS")
+    result_file = open(result_path, "w")
+    kvm_ver = utils.system_output(kvm_ver_chk_cmd)
+    host_ver = os.uname()[2]
+    result = session.cmd_output(guest_ver_cmd, timeout)
+    guest_ver = re.findall(".*?(\d{2}\.\d{2}\.\d{3}\.\d{4}).*?", result)
+    result_file.write("### kvm-userspace-ver : %s\n" % kvm_ver)
+    result_file.write("### kvm_version : %s\n" % host_ver)
+    result_file.write("### guest-kernel-ver :%s" % result)
+
+    # get record_list
+    record_line = ""
+    for record in record_list.split():
+        record_line += "%s|" % format_result(record)
+
+    #get result tested by each scenario
+    for pkt_cate in category.split():
+        result_file.write("Category:%s\n" % pkt_cate)
+        result_file.write("%s\n" % record_line.rstrip("|"))
+
+        # copy pktgen_test script to test server
+        local_path = os.path.join(data_dir.get_root_dir(),
+                                  "shared/scripts/pktgen_perf.sh")
+        remote_path = "/tmp/pktgen_perf.sh"
+        if pkt_cate == "tx":
+            pktgen_ip = vm.wait_for_get_address(0, timeout=5)
+            remote.scp_to_remote(pktgen_ip, shell_port, username,
+                                 password, local_path, remote_path)
+        elif pkt_cate == "rx":
+            process.run("cp %s %s" % (local_path, remote_path))
+
+        for size in pkt_size.split():
+            if pkt_cate == "tx":
+                error_context.context("test guest tx pps performance",
+                                      logging.info)
+                guest_mac = vm.get_mac_address(0)
+                pktgen_interface = utils_net.get_linux_ifname(session,
+                                                              guest_mac)
+                dsc_dev = utils_net.Interface(vm.get_ifname(0))
+                dsc_mac = dsc_dev.get_mac()
+                dsc = dsc_mac
```

Comment:
dsc_mac = dsc_dev.get_mac()
dsc = dsc_mac
Why don't you write them in "dsc = dsc_dev.get_mac()" ?

Reply:
I have updated. please have a check, thanks.